lib/Listener/SyncGroupListener.php

```diff
@@ -0,0 +1,180 @@
+     * @throws \Xibo\Support\Exception\NotFoundException
+     */
+    public function onUserDelete(UserDeleteEvent $event, $eventName, EventDispatcherInterface $dispatcher)
+    {
+        $user = $event->getUser();
+        $function = $event->getFunction();
+        $newUser = $event->getNewUser();
+
+        if ($function === 'delete') {
+            // we do not want to delete Display specific Display Groups, reassign to systemUser instead.
+            foreach ($this->syncGroupFactory->getByOwnerId($user->userId) as $syncGroup) {
+                $syncGroup->delete();
+            }
+        } else if ($function === 'reassignAll') {
+            foreach ($this->syncGroupFactory->getByOwnerId($user->userId) as $syncGroup) {
+                $syncGroup->setOwner($newUser->getOwnerId());
+                $syncGroup->save();
+            }
+        } else if ($function === 'countChildren') {
+            $syncGroups = $this->syncGroupFactory->getByOwnerId($user->userId);
+
+            $count = count($syncGroups);
+            $this->getLogger()->debug(
+                sprintf(
+                    'Counted Children Sync Groups on User ID %d, there are %d',
+                    $user->userId,
+                    $count
+                )
+            );
+
+            $event->setReturnValue($event->getReturnValue() + $count);
+        }
+    }
+
+    /**
+     * @param ParsePermissionEntityEvent $event
+     * @return void
+     * @throws \Xibo\Support\Exception\NotFoundException
+     */
+    public function onParsePermissions(ParsePermissionEntityEvent $event)
+    {
+        $this->getLogger()->debug('onParsePermissions');
+        $event->setObject($this->syncGroupFactory->getById($event->getObjectId()));
+    }
+
+    /**
+     * @param FolderMovingEvent $event
+     * @return void
+     * @throws \Xibo\Support\Exception\NotFoundException
+     */
+    public function onFolderMoving(FolderMovingEvent $event)
+    {
+        $folder = $event->getFolder();
+        $newFolder = $event->getNewFolder();
+
+        foreach ($this->syncGroupFactory->getByFolderId($folder->getId()) as $syncGroup) {
+            $syncGroup->folderId = $newFolder->getId();
+            $syncGroup->permissionsFolderId = $newFolder->getPermissionFolderIdOrThis();
+            $syncGroup->updateFolders('syncgroup');
+        }
+    }
+
+    /**
+     * @param ScheduleSaveEvent $event
+     * @return void
+     * @throws NotFoundException
+     */
+    public function onScheduleSave(ScheduleSaveEvent $event): void
+    {
+        $scheduleEvent = $event->getEvent();
+        $params = $event->getParams();
+        if ($scheduleEvent->eventTypeId === Schedule::$SYNC_EVENT) {
+            $syncGroup = $this->syncGroupFactory->getById($scheduleEvent->syncGroupId);
+
+            foreach ($syncGroup->getSyncGroupMembers() as $display) {
+                $this->storageService->insert('INSERT INTO `schedule_sync` (`eventId`, `displayId`, `layoutId`)
```

Comment:
I think we usually do this type of insert in the entity - so in this case the schedule entity. I'd perhaps expect this to be in `schedule->save()`